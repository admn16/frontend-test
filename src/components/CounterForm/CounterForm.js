import React from 'react';
import styled from 'styled-components';
import Card from 'components/Card/Card';

const CounterForm = () => {
  return (
    <Card>
      <form>
        <input />
        <button>+</button>
      </form>
    </Card>
  );
};

export default CounterForm;