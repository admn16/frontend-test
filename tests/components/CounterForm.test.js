import React from 'react';
import renderer from 'react-test-renderer';
import CounterForm from 'components/CounterForm/CounterForm';

describe('<CounterForm/>', () => {
  const props = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    text: 'Title',
  };

  test('renders correctly', () => {
    const tree = renderer
      .create(<CounterForm {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
