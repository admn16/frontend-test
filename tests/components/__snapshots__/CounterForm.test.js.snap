// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterForm/> renders correctly 1`] = `
<div
  className="Card__StyledCard-hu4gdg-0 hNyZiT"
>
  <form
    className="CounterForm__Form-sc-1w9o0z7-0 emKkNx"
    onSubmit={[MockFunction]}
  >
    <input
      className="CounterForm__Input-sc-1w9o0z7-1 fHQEKr"
      onChange={[MockFunction]}
      placeholder="Add new counter"
      value="Title"
    />
    <button
      className="CounterForm__Button-sc-1w9o0z7-2 iXLQka"
      type="submit"
    >
      +
    </button>
  </form>
</div>
`;
